pkgs/development/interpreters/python/default.nix

```diff
@@ -111,6 +111,38 @@ in {
     inherit passthruFun;
   };
 
+  # Minimal versions of Python (built without optional dependencies)
+  python3Minimal = (callPackage ./cpython {
+    self = python3Minimal;
+    sourceVersion = {
+      major = "3";
+      minor = "7";
+      patch = "4";
+      suffix = "";
+    };
+    sha256 = "0gxiv5617zd7dnqm5k9r4q2188lk327nf9jznwq9j6b8p0s92ygv";
+    inherit (darwin) CF configd;
+    inherit passthruFun;
+
+    # MINIMIZE ALL THE THINGS!!
+    openssl = null;
+    readline = null;
+    ncurses = null;
+    gdbm = null;
+    sqlite = null;
+    stripConfig = true;
```

Comment:
Does this prevent building c extensions or embed this version of python in some projects depending on `python-config`?

Reply:
yes. `python-config` from the passed python would be used to get include and linker flags, and as it doesn't exist, you won't be able to use it.

Reply:
Sidenote: It seems you only need `python-config` if you want to reuse that version of python in another project, but building a `python3Minimal.withPackages` calling some C  bindings works just fine.